### PR TITLE
Add missing metrics from go client (#379)

### PR DESCRIPTION
**Pull request by aai95**
Metric `decision-task-force-completed` was added
Metric `non-deterministic-error` was added

### CODE REVIEW

**Comment by aai95** [comment]
@vancexu could you please make a review?

**Comment by aai95** [comment]
Dear @mfateev could you please review this PR?

File: src/main/java/com/uber/cadence/internal/replay/DecisionsHelper.java [hunk 1]
```diff
@@ -716,8 +723,12 @@ private boolean addMissingVersionMarker(
   }
 
   private DecisionStateMachine getDecision(DecisionId decisionId) {
+    Scope metricsScope = options
```
Comment (meiliang86) [inline src/main/java/com/uber/cadence/internal/replay/DecisionsHelper.java:726]: Nit: move it inside the if statement?

File: src/test/java/com/uber/cadence/internal/replay/ReplayDeciderCacheTests.java [hunk 2]
```diff
@@ -254,6 +254,7 @@ private ReplayDecider doNotCreateFakeDecider(
   }
 
   private ReplayDecider createFakeDecider(PollForDecisionTaskResponse response) {
+    SingleWorkerOptions options = mock(SingleWorkerOptions.class);
```
Comment (meiliang86) [inline src/test/java/com/uber/cadence/internal/replay/ReplayDeciderCacheTests.java:257]: This is causing issues for unit tests as SingleWorkerOptions is final and cannot be mocked. Just manually build a SingleWorkflowOptions instead.
```
com.uber.cadence.internal.replay.ReplayDeciderCacheTests > whenHistoryIsFullNewReplayDeciderIsReturnedAndCached_InitiallyEmpty FAILED

    org.mockito.exceptions.base.MockitoException: 

    Cannot mock/spy class com.uber.cadence.internal.worker.SingleWorkerOptions

    Mockito cannot mock/spy following:

      - final classes

      - anonymous classes

      - primitive types

        at com.uber.cadence.internal.replay.ReplayDeciderCacheTests.createFakeDecider(ReplayDeciderCacheTests.java:257)

        at com.uber.cadence.internal.replay.ReplayDeciderCacheTests.lambda$whenHistoryIsFullNewReplayDeciderIsReturnedAndCached_InitiallyEmpty$0(ReplayDeciderCacheTests.java:64)

        at com.uber.cadence.internal.replay.DeciderCache.getOrCreate(DeciderCache.java:65)

        at com.uber.cadence.internal.replay.ReplayDeciderCacheTests.whenHistoryIsFullNewReplayDeciderIsReturnedAndCached_InitiallyEmpty(ReplayDeciderCacheTests.java:64)
```

**Comment by meiliang86** [comment]
Thanks for your contribution @aai95 !

File: src/main/java/com/uber/cadence/internal/replay/DecisionsHelper.java (hunk 1 above)
Comment (aai95) [reply src/main/java/com/uber/cadence/internal/replay/DecisionsHelper.java]: Done

File: src/test/java/com/uber/cadence/internal/replay/ReplayDeciderCacheTests.java (hunk 2 above)
Comment (aai95) [reply src/test/java/com/uber/cadence/internal/replay/ReplayDeciderCacheTests.java]: Done

**Comment by aai95** [comment]
@meiliang86 should I fix something else or we just wait for review?

**Comment by aai95** [comment]
Thank you @meiliang86 
I think issue https://github.com/uber/cadence-java-client/issues/379 can be closed

**Pull request merged**